### PR TITLE
Bratseth/persist metrics rebased

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java [hunk 1]
```diff
@@ -53,41 +57,69 @@ public void activate(Collection<HostSpec> hosts, NestedTransaction transaction,
...
+        rememberResourceChange(transaction, generation,
+                               NodeList.copyOf(oldActive).not().retired(),
+                               NodeList.copyOf(newActive).not().retired());
         unreserveParentsOf(reservedToActivate);
     }
 
+    private void rememberResourceChange(ApplicationTransaction transaction, long generation,
+                                        NodeList oldNodes, NodeList newNodes) {
+        if (nodeRepository.applications().get(transaction.application()).isEmpty()) return; // infrastructure app, hopefully
+        Application application = nodeRepository.applications().get(transaction.application()).get();
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java:102]: Read this once from ZK?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeList.java [hunk 2]
```diff
@@ -179,6 +180,13 @@ public NodeList group(int index) {
                                                        .findFirst());
     }
 
+    public ClusterResources toResources() {
+        if (isEmpty()) return new ClusterResources(0, 0, NodeResources.unspecified());
+        return new ClusterResources(size(),
+                                    (int)stream().distinct().count(),
```
Comment (freva) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeList.java:186]: Shouldn't there be a map to to group id here?

**Review by freva — APPROVED** [review]

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java (hunk 1 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java:102]: This is reading it once?

Comment (freva) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java:102]: It's reading it in the if statement above as well?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeList.java (hunk 2 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeList.java]: Yes, thanks!

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java (hunk 1 above)
Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/Activator.java]: That is true. Changed.

**Pull request merged**